packages/abi-typegen/test/fixtures/forc-projects/minimal-with-configurable/src/main.sw

```diff
@@ -1,7 +1,17 @@
 contract;
 
+struct GenericStruct<T1, T2> {
+    p1: T1,
+    p2: T2,
+}
+
 configurable {
     SHOULD_RETURN: bool = true,
+    AN_OPTION: Option<u8> = Option::None,
+    A_GENERIC_STRUCT: GenericStruct<GenericStruct<u8, u16>, u32> = GenericStruct {
+        p1: GenericStruct { p1: 4, p2: 257 },
+        p2: 57000,
+    },
```

Comment:
The `Option<T>` is a generic enum.

Reply:
They generate different results though - do they not?

Reply:
I'm not sure what you mean. An `Option<T>` enum is treated the same way as any other generic enum and its generic parameters (`Option<BigNumberish>` in this case) are specified correctly.

Reply:
```sway
enum GenericEnum<V> {
    Foo: V,
}
```

Should generate this should it not? Which is different from an `Option<T>`?

```ts
export type GenericEnumInput<V> = Enum<{ Foo: V }>;
export type GenericEnumOutput<V> = GenericEnumInput<V>;
```

I'm happy with the implementation so I'm approving - just curious about what you mean.

Reply:
Aha, I see what you mean. You're right, but I'm okay with things as they are here because the logic for specifying the generic parameters is shared between all of these types anyways and the "complex" struct example is enough.